chocolatey/choco
# (#3477) Replace Get-ChecksumValid with Assert-ValidChecksum cmdlet.

## Description Of Changes
- Remove `Get-ChecksumValid`
- Add `Assert-ValidChecksum`
- Add tests for command to verify behaviour matches the old command.
- Add alias for backwards compatibility with packages.
- Minor fixes for the doc-gen script to address issues I ran into.
- Added a framework we can use in future for deprecating old aliases or to-be-removed commands in a simple and standardised way.

This PR _does not_ contain the compiled XML help information for the command. In discussion with @gep13 we agreed that it would be better to compile that in the final release. I will have a docs PR linked here in a moment, watch this space.

**Note**: As this PR is part of #3477 and is thus a breaking change to a degree, we will need to cut a `support/2.x` branch once this is merged, so that none of the breaking changes associated with #3477 inadvertently make their way into the 2.x versions.

## Motivation and Context
See #3477 

Documentation PR: https://github.com/chocolatey/docs/pull/1075

## Testing

Tested locally with the accompanying Pester tests:
1. From an administrative PowerShell session
2. Run `.\build.ps1`
3. Run `.\Invoke-Tests.ps1 -Tag AssertValidChecksum`
4. From VS, run the integration tests and verify there are no unexpected issues.

### Proving out the deprecated commands framework

After running the above, close and reopen the admin PowerShell session, and then follow these steps:
1. Apply the following patch: [Get-ValidCheckSum-deprecation-verification.patch](https://github.com/user-attachments/files/17262230/Get-ValidCheckSum-deprecation-verification.patch)
2. Rerun the prior steps to run the build and tests again, including the code and test change.
3. Monitor the output from the tests; near the start of the test run, you should see a warning issued about the test calling `Get-ChecksumValid` and that the alias is deprecated and should be replaced with `Assert-ValidChecksum`

### Operating Systems Testing

Windows 10

## Change Types Made



## Change Checklist


## Related Issue


Part of #3477
